azure-pipelines.yml

```diff
@@ -40,7 +40,14 @@ steps:
 - script: dotnet build --configuration $(buildConfiguration)
   displayName: 'dotnet build $(buildConfiguration)'
 
-- script: dotnet-sonarscanner end /d:sonar.login="$(sonar.token)"
+- task: DotNetCoreCLI@2
+  displayName: 'dotnet test $(buildConfiguration)'
+  inputs:
+    command: test
+    projects: '**/*.sln'
+    arguments: '--configuration $(buildConfiguration) --collect "Code coverage"'
```

Comment:
Add the argument `--no-build` to this line.
Without this argument, `dotnet test` will build the solution again.

This was enough to make the build complete and generate code coverage for me. I've no idea why the building the projects multiple times causes the scanner to produce the message _No analyzable projects found_; the scanner should handle multiple builds. I'll have to investigate that further.


The coverage appears in Azure DevOps, but isn't uploaded to SonarCloud - that's a separate issue.